components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/test/java/org/wso2/carbon/identity/role/v2/mgt/core/dao/RoleDAOTest.java

```diff
@@ -982,6 +990,147 @@ void testIsValidSubOrgPermission() throws Exception {
         assertFalse((Boolean) isValidSubOrgPermissionMethod.invoke(roleDAO, "console:applications"));
     }
 
+    @Test
+    public void testGetSharedRoleToMainRoleMappingsBySubOrgWithSharedRoleUUID() throws Exception {
+
+        RoleDAOImpl roleDAO = setupRoleDaoImpl();
+
+        // Adding main roles and shared roles
+        RoleBasicInfo mainRole = addRole(MAIN_ROLE_NAME, APPLICATION_AUD, SAMPLE_APP_ID, roleDAO);
+        RoleBasicInfo sharedRole = addRole(SHARED_ROLE_NAME, APPLICATION_AUD, SAMPLE_APP_ID, roleDAO);
+        roleDAO.addMainRoleToSharedRoleRelationship(mainRole.getId(), sharedRole.getId(), SAMPLE_TENANT_DOMAIN,
+                SAMPLE_TENANT_DOMAIN);
```

Comment:
```suggestion
        RoleBasicInfo mainRole = addRole(MAIN_ROLE_NAME, APPLICATION_AUD, SAMPLE_APP_ID, roleDAO);
        RoleBasicInfo sharedRole = addRole(SHARED_ROLE_NAME, APPLICATION_AUD, SAMPLE_APP_ID, roleDAO);
        roleDAO.addMainRoleToSharedRoleRelationship(mainRole.getId(), sharedRole.getId(), SAMPLE_TENANT_DOMAIN,
                SUB_ORG_TENANT_DOMAIN);
```

Reply:
Addressed and resolved in commit: [cccc26e](https://github.com/wso2/carbon-identity-framework/pull/6162/commits/cccc26ee039e448196fc1299e2827453c1bc76b5)